core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go

```diff
@@ -1373,8 +1373,14 @@ func TestEthBroadcaster_ProcessUnstartedEthTxs_Errors(t *testing.T) {
 		// In this scenario the node operator REALLY fucked up and set the bump
 		// to zero (even though that should not be possible due to config
 		// validation)
+		oldGasBumpWei := evmcfg.EvmGasBumpWei()
+		oldGasBumpPercent := evmcfg.EvmGasBumpPercent()
 		cfg.Overrides.GlobalEvmGasBumpWei = big.NewInt(0)
 		cfg.Overrides.GlobalEvmGasBumpPercent = null.IntFrom(0)
+		defer func() {
+			cfg.Overrides.GlobalEvmGasBumpWei = oldGasBumpWei
+			cfg.Overrides.GlobalEvmGasBumpPercent = null.IntFrom(int64(oldGasBumpPercent))
+		}()
```

Comment:
Is this resetting shared state from the parent test? Worth a comment at least?
Would `t.Cleanup` be more or less reliable? :thinking: maybe only different in irrelevant situations anyways...

Reply:
Yes it's resetting state. `t.Cleanup` might be ambiguous, is it this `t` or the parent one.

Reply:
IMHO it reads pretty clearly, but actually from a higher level, is it even safe to do any mutations on cfg from subtests? Once the cfg is in use, these are all racy, aren't they?

Reply:
Could subset tests run in parallel?

Reply:
Only if explicitly marked as Parallel, but they all use long running concurrent components from the _parent_ test that could race to read these overrides.